opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -189,27 +186,40 @@ namespace Opm {
        return nodes;
    }
 
-    std::vector< ERT::smspec_node > handleKW( const DeckKeyword& keyword, const EclipseState& es ) {
+    std::vector< ERT::smspec_node > handleKW( const DeckKeyword& keyword,
+                                              const Schedule& schedule,
+                                              const Eclipse3DProperties& props,
+                                              std::array< int, 3 > n_xyz ) {
         const auto var_type = ecl_smspec_identify_var_type( keyword.name().c_str() );
 
         switch( var_type ) {
             case ECL_SMSPEC_WELL_VAR: /* intentional fall-through */
-            case ECL_SMSPEC_GROUP_VAR: return keywordWG( var_type, keyword, es );
-            case ECL_SMSPEC_FIELD_VAR: return keywordF( keyword, es );
-            case ECL_SMSPEC_BLOCK_VAR: return keywordB( keyword, es );
-            case ECL_SMSPEC_REGION_VAR: return keywordR( keyword, es );
-            case ECL_SMSPEC_COMPLETION_VAR: return keywordC( keyword, es );
+            case ECL_SMSPEC_GROUP_VAR: return keywordWG( var_type, keyword, schedule );
+            case ECL_SMSPEC_FIELD_VAR: return keywordF( keyword );
+            case ECL_SMSPEC_BLOCK_VAR: return keywordB( keyword, n_xyz );
+            case ECL_SMSPEC_REGION_VAR: return keywordR( keyword, props, n_xyz );
+            case ECL_SMSPEC_COMPLETION_VAR: return keywordC( keyword, schedule, n_xyz );
 
             default: return {};
         }
     }
 
-    SummaryConfig::SummaryConfig( const Deck& deck, const EclipseState& es ) {
+    SummaryConfig::SummaryConfig( const Deck& deck, const EclipseState& es )
+        : SummaryConfig( deck,
+                         *es.getSchedule(),
+                         es.get3DProperties(),
+                         dimensions( *es.getInputGrid() ) )
+    {}
+
+    SummaryConfig::SummaryConfig( const Deck& deck,
+                                  const Schedule& schedule,
+                                  const Eclipse3DProperties& props,
```

Comment:
We should augment the `Eclipse3DProperties` class with the `(nx,ny,nz)` information, that way we can shorten this signature slightly and it makes a certain sense to have there as well.